lib/outbox_repository.py

```
@@ -0,0 +1,166 @@
+import json
+
+# from typing import Union, Dict, Any
+
+# Try to import Flask dependencies, fallback to basic logging if not available
+try:
+    from app.logging import get_logger
+    from app.models.database import db
+    from app.models.outbox import Outbox
+    from lib.db import session_guard
+
+    logger = get_logger(__name__)
+    FLASK_AVAILABLE = True
+except ImportError as e:
+    import logging
+
+    logging.basicConfig(level=logging.DEBUG)
+    logger = logging.getLogger(__name__)
+    logger.warning(f"Flask dependencies not available: {e}")
+    FLASK_AVAILABLE = False
+
+
+def _create_update_event_payload(event_dict):
```

Should include a return type hint. Also, I think it would make more sense if this function raised an error rather than return None when a required field is missing.